sapirbitan/assignment2/ex1.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+using namespace std;
+
+struct node {
+	int key;
+	node* left_child = NULL;
+	node* right_child = NULL;
+	node* ancestor = NULL;
+};//assuming every node contains pointers to it's children and ancestors
+  //(the root's ancestor will point to NULL)
+
+// this function finds the key which we want to print it's ancestors from
+//it goes over all of the tree and returns the node we were looking for
+node* find_key(node& root, int& key) {
+	node* found_key = NULL;
+	if (&root == NULL) 
+		return NULL	;
```

Comment:
Please be consistent in your formatting, this line should be (without extra space between 'NULL' and ';'):

`return NULL;`